src/audio/module_adapter/module/dts.c

```diff
@@ -350,15 +347,16 @@ static int dts_codec_apply_config(struct processing_module *mod)
 			/* Calculate size of param->data */
 			param_data_size = param->size - param_header_size;
 
+			comp_dbg(dev, "dts_codec_apply_config() id %d size %d",
+					param->id, param_data_size);
+
 			if (param_data_size) {
 				dts_result = dtsSofInterfaceApplyConfig(codec->private, param->id,
 					param->data, param_data_size);
-				ret = dts_effect_convert_sof_interface_result(dev, dts_result);
-				if (ret) {
+				if (dts_result != DTS_SOF_INTERFACE_RESULT_SUCCESS) {
 					comp_err(dev,
 						"dts_codec_apply_config() dtsSofInterfaceApplyConfig failed %d",
 						dts_result);
```

Comment:
why is it changed to suppress error on apply config?

Reply:
As we've integrated against various platforms, we've encountered a few occurrences where invalid configuration data is sent at startup, which we would detect and report to the module adapter, which then results in the DSP not starting, causing delays in the integration effort.
I think it's better here for us to absorb the error and create a log message.

Reply:
I believe the invalid config at startup is caused by the outdated topology which still have setup and runtime config controls for module_adapter. The module_adapter c code had removed setup config, so those bytes fed by old topology will be passed here and causes errors.

Reply:
Yes. This is one of the scenarios. And I think a good example of the motivation for this change.
As the module adapter continues to change shape on the mainstream branch and those changes get distributed to some of the per-platform branches, I think we (DTS) struggle to understand when, where and, sometimes, how we should make corresponding changes to our topology files. This change isolates us from that - we now only need to confirm that our configuration blob delivered via UCM is applied successfully, and know that all other blobs we don't care about will be quietly rejected is they aren't suitable

Reply:
I beg to differ. It is indeed a hard task to maintain the module working on multiple production branches. We are taking progress on migrating the elder branch (e.g. adl-004-drop-stable) to more up-to-date one (rpl-001-drop-stable) to alleviate the code difference from main. Even though the varied designs are required on different branches, we should still keep the correct one on mainstream.

From another aspect, the suppressed failure could be a potential landmine which causes the fatal error at the latter timestamp. It will make debugging more difficult because the SOF log is not achievable from the normal (official) OS images (we need to enable debug_trace feature in kernel config and build/deploy kernel). We should raise the error promptly unless we have fair confidence the error is harmless.

Reply:
@johnylin76 
I've already reverted this part to previous one. Please review. Thanks.